plugins/outputs/nats/nats.go

```diff
@@ -85,8 +132,51 @@ func (n *NATS) Connect() error {
 
 	// try and connect
 	n.conn, err = nats.Connect(strings.Join(n.Servers, ","), opts...)
+	if err != nil {
+		return err
+	}
+
+	if n.Jetstream != nil {
+		n.jetstreamClient, err = jetstream.New(n.conn)
+		if err != nil {
+			return fmt.Errorf("failed to connect to jetstream: %w", err)
+		}
+
+		if len(n.Jetstream.Subjects) == 0 {
+			n.Jetstream.Subjects = []string{n.Subject}
+		}
+		if !choice.Contains(n.Subject, n.Jetstream.Subjects) {
+			n.Jetstream.Subjects = append(n.Jetstream.Subjects, n.Subject)
+		}
+		var streamConfig jetstream.StreamConfig
+		n.convertToJetstreamConfig(&streamConfig)
+		_, err = n.jetstreamClient.CreateOrUpdateStream(context.Background(), streamConfig)
+		if err != nil {
+			return fmt.Errorf("failed to create or update stream: %w", err)
+		}
+		n.Log.Infof("stream (%s) successfully created or updated", n.Jetstream.Name)
```

Comment:
Shouldn't this be a debug message? In any case it should start with an uppercase letter...